### ISSUE TITLE
Limit user to total Plays/Watches per day

### ISSUE THREAD

**Issue by Anuruddha094** [blacktwin/JBOPS#218]
[https://github.com/blacktwin/JBOPS/blob/master/killstream/limiterr.py](https://github.com/blacktwin/JBOPS/blob/master/killstream/limiterr.py)

**It's pretty handy if we can limit users  based on how many watches/plays per day.** _(This script already can limit  play sessions per day)_
**As an example a user can watch any 5 media per day as much as they want. So they can start, resume and stop and start again same 5 content. Once 24 hours gone they can watch another 5 content or unlock one by one according to start time of watched sequence.**

It pretty much like same syntax as Limit user to total play sessions per day. But should be like this as my opinion.

`--jbop watch --username {username} --sessionId {session_id} --days 0 --limit plays=5 --killMessage "You have met your limit of 5 watch per day."`

cheers

**Comment by blacktwin** [comment]
I think the parts are all there, code wise, you'd just have to add the check for the individual item instead of just the Show level (see:[`todo`](https://github.com/blacktwin/JBOPS/blob/master/killstream/limiterr.py#L359-L360)). You'd also want to change your arguments as what you've provided is already [used](https://github.com/blacktwin/JBOPS/blob/master/killstream/limiterr_readme.md#limit-user-to-total-playswatches-and-send-a-notification-to-agent-1). Feel free edit to your needs. If you'd like to create a PR with your changes, please feel free.

I do not understand the use case. Why restrict someone to only X things but they can watch those X things an unlimited amount of times? Using Labels might be better.

**Comment by Anuruddha094** [comment]
> I think the parts are all there, code wise, you'd just have to add the check for the individual item instead of just the Show level (see:[`todo`](https://github.com/blacktwin/JBOPS/blob/master/killstream/limiterr.py#L359-L360)). You'd also want to change your arguments as what you've provided is already [used](https://github.com/blacktwin/JBOPS/blob/master/killstream/limiterr_readme.md#limit-user-to-total-playswatches-and-send-a-notification-to-agent-1). Feel free edit to your needs. If you'd like to create a PR with your changes, please feel free.
> 
> I do not understand the use case. Why restrict someone to only X things but they can watch those X things an unlimited amount of times? Using Labels might be better.

yes script already has all parts but we need some tweaks. I'm not a Python programmer but I will give it a try.

**Achievements** 

- Stop users are watching more than X number of media from any library for a day.
- Let them to play same (what ever played within same 24 hours) X number of media, unlimited  times for a day (Start, Stop, Pause and Play has no restrictions).
- After 24 hours of the first media play time, let them to watch X number of media again. 
 _(It's nice if we can unlock each media based on start time once it reached 24 hours)_


**Use case**

Basically this will help, if somebody wants to restrict users keep playing different media per day without actually watching anything completely, In a scenario user just play a movie, fast forward it and stop it. And play another new movie do the same thing and keep repeating it unlimited times. This will add unnecessary stress for the server if its a transcode stream.

cheers
